src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -802,14 +802,19 @@ private static bool IsInstance(CallingConventions callingConvention) =>
 
         internal static SignatureCallingConvention GetSignatureConvention(CallingConventions callingConvention)
         {
-            SignatureCallingConvention convention = SignatureCallingConvention.Default;
+            SignatureCallingConventionEx convention = (SignatureCallingConventionEx)SignatureCallingConvention.Default;
 
             if ((callingConvention & CallingConventions.VarArgs) != 0)
             {
-                convention = SignatureCallingConvention.VarArgs;
+                convention = (SignatureCallingConventionEx)SignatureCallingConvention.VarArgs;
             }
 
-            return convention;
+            if ((callingConvention & CallingConventions.HasThis) != 0)
+            {
+                convention |= SignatureCallingConventionEx.HasThis;
+            }
```

Comment:
Should this handle `CallingConventions.ExplicitThis` as well?

Reply:
Yes, thanks. We may need it shortly.